clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -2816,7 +2818,18 @@ int main(int Argc, char **Argv) {
     timeTraceProfilerInitialize(Granularity, Argv[0]);
   }
 
+  UseSYCLPostLinkTool = !Args.hasArg(OPT_no_use_sycl_post_link_tool);
```

Comment:
Could we use `Args.hasFlag` instead? I think that usually the pattern with the positive/negative options.

Reply:
The line `UseSYCLPostLinkTool = Args.hasArg(OPT_use_sycl_post_link_tool);` doesn't have an effect at the moment since `-use-sycl-post-link-tool` is turned on by default. Only presence of `-no-use-sycl-post-link-tool` has the effect. It will change after switching to the SYCL library.

I can change the code to the following:
```cpp
UseSYCLPostLinkTool = true;
if (Args.hasArg(OPT_use_sycl_post_link_tool) && Args.hasArg(OPT_no_use_sycl_post_link_tool))
  reportError("both options can't be used");

if (Args.hasArg(OPT_no_use_sycl_post_link_tool))
  UseSYCLPostLinkTool = false;
```

How does it look to you?

Reply:
It's not a huge deal but ideally Clang would always pass one of the two options, so we can use `hasFlag` and implement the default there, but for now the origin code you had is fine.

Reply:
Done that.